### ISSUE TITLE
nvim_feedkeys not blocking as expected?

### ISSUE THREAD

**Issue by Grazfather** [neovim/neovim#16571]
### Neovim version (nvim -v)

v0.7.0-dev+653-g7229afcd6

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

macOS

### Terminal name/version

N/A

### $TERM environment variable

N/A

### Installation

brew

### How to reproduce the issue

Bind `"<cmd>lua vim.api.nvim_feedkeys('A', 'n', true)<CR>jjj<ESC>"` and then execute it from normal mode.

### Expected behavior

Expected (I think?): Append "jjj" at the end of the current line.

### Actual behavior

Go down 3 lines then enter insert mode.

Basically the `A` at I input doesn't seem to change the mode into insert mode by the time the `jjj` input is run, so it's run as if in normal mode.

**Comment by Grazfather** [comment]
That was a copy and paste error, I was testing out both functions. Edited.

**Comment by zeertzjq** [comment]
Change `'n'` to `'ni'`.

```
<cmd>lua vim.api.nvim_feedkeys('A', 'ni', true)<CR>jjj<ESC>
```

When the mapping is processed, all of the mapping is appended to the typeahead buffer. When the `<Cmd>` part is executed and `feedkeys` uses mode `'n'`, `'A'` is appended to the typeahead buffer, which is after the `jjj<ESC>`. Use `feedkeys` mode `'ni'` to insert it to the beginning of the typeahead buffer.

**Comment by Grazfather** [comment]
Thank you. That makes sense.
